modules/apps/portal/portal-dao-orm-custom-sql-impl/src/main/java/com/liferay/portal/dao/orm/custom/sql/internal/CustomSQLImpl.java

```diff
@@ -902,6 +896,30 @@ private CustomSQLContainer _getCustomSQLContainer(ClassLoader classLoader) {
 		return new CustomSQLContainer(classLoader, sourceURL);
 	}
 
+	private String _getStatusesSQL(QueryDefinition<?> queryDefinition) {
+		String prefix = "";
+		String statusCondition;
+
+		if (queryDefinition.isExcludeStatuses()) {
+			statusCondition = _STATUS_CONDITION_INVERSE;
+		}
+		else {
+			statusCondition = _STATUS_CONDITION_DEFAULT;
+		}
+
+		StringBundler sb = new StringBundler();
+
+		int[] statuses = queryDefinition.getStatuses();
+
+		for (int i = 0; i < statuses.length; i++) {
```

Comment:
We iterate over the statuses and prepend them with either `" AND "` or `" OR "`. Initializing `prefix` to the empty string makes it so that the first iteration is well-formed.